app/assets/javascripts/views/schedule.js

```diff
@@ -53,6 +54,17 @@ Yacs.views.schedule = function (data) {
     crnListElement.textContent = 'CRNs: ' + scheduleData.crns.join(', ');
   }
 
+  /* this is before `if(data.schedules.length==0)` because clear selection should */
+  /* still work even if courses conflict and there are zero possible schedules    */
+  Yacs.on('click', clearButtonElement, function () {
+    /* clear if the user has any selections */  
+    if (Yacs.user.getSelections().length != 0) {
+      Yacs.user.clearSelections();
+      Yacs.views.schedule([]);
```

Comment:
has this been vetted? this should cause an error. the top-level object is still necessary. it should be: `{ schedules: [] }`